modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala

```diff
@@ -77,6 +77,10 @@ object Execution {
 sealed trait Result
 
 object Result {
-  case class OK[R](r: R) extends Result
+  case class OK(response: Response) extends Result
   case class Error[E](e: E) extends Result
+
+  sealed trait Response
+  case class Configured(r: String) extends Response
+  case class Observed(fileId: FileId) extends Response
```

Comment:
Actually I want to look there, because  the observation buttons should be displayed only when the observation begins, so it is kind of a marker event. It would be ok to the fileId in the step though